src/modules/silentpayments/main_impl.h

```diff
@@ -365,4 +365,349 @@ int secp256k1_silentpayments_recipient_create_labeled_spend_pubkey(const secp256
     return 1;
 }
 
+/** A explanation of the prevouts_summary object and its usage:
+ *
+ *  The prevouts_summary object contains:
+ *
+ *  [magic: 4 bytes][boolean: 1 byte][prevouts_pubkey_sum: 64 bytes][input_hash: 32 bytes]
+ *
+ *  The magic bytes are checked by functions using the prevouts_summary object to
+ *  check that the prevouts_summary object was initialized correctly.
+ *
+ *  The boolean (combined) indicates whether or not the summed prevout public keys and the
+ *  input_hash scalar have already been combined or are both included. The reason
+ *  for keeping input_hash and the summed prevout public keys separate is so that an elliptic
+ *  curve multiplication can be avoided when creating the shared secret, i.e.,
+ *  (recipient_scan_key * input_hash) * prevout_pubkey_sum.
+ *
+ *  But when storing the prevouts_summary object, either to send to light clients or for
+ *  wallet rescans, we can save 32-bytes by combining the input_hash and prevout_pubkey_sum and saving
+ *  the resulting point serialized as a compressed public key, i.e., input_hash * prevout_pubkey_sum.
+ *
+ *  For each function:
+ *
+ *  - `_recipient_prevouts_summary_create` always creates a prevouts_summary object with combined = false
+ *  - `_recipient_prevouts_summary_serialize` multiplies the input_hash into the summed public key before
+ *     serializing the resulting point as a compressed public key, if combined = false. If combined = true,
+ *     the point is serialized back into a compressed public key.
+ *  - `_recipient_prevouts_summary_parse` assumes the input represents a previously serialized
+ *    prevouts_summary object and always deserializes into a prevouts_summary object with combined = true
+ *    (and the input_hash portion zeroed out).
+ */
+
+int secp256k1_silentpayments_recipient_prevouts_summary_create(
+    const secp256k1_context *ctx,
+    secp256k1_silentpayments_prevouts_summary *prevouts_summary,
+    const unsigned char *outpoint_smallest36,
+    const secp256k1_xonly_pubkey * const *xonly_pubkeys,
+    size_t n_xonly_pubkeys,
+    const secp256k1_pubkey * const *plain_pubkeys,
+    size_t n_plain_pubkeys
+) {
+    size_t i;
+    size_t pubkeylen = 64;
```

Comment:
nit: other functions that access the `_prevouts_summary.data` array (e.g. `_prevouts_summary_{serialize,parse}`) don't use this constant and directly access at offset `5 + 64` instead, so for consistency, could either remove it here (my preference) or introduce it in the other functions as well